app/services/notifications/sync_system_notifications_service.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+module Notifications
+  class SyncSystemNotificationsService
+    extend Service
+    include HTTParty
+    base_uri Rails.application.secrets.system_notifications_uri
+
+    attr_reader :errors
+
+    def initialize
+      @errors = {}
+    end
+
+    def call
+      call_api
+
+      save_new_notifications if succeed?
+
+      self
+    end
+
+    def succeed?
+      @errors.none?
+    end
+
+    private
+
+    def call_api
+      last_sync = SystemNotification.last_sync_timestamp
+      channel = Rails.application.secrets.system_notifications_channel
+
+      unless last_sync
+        @errors[:last_sync_timestamp] = 'Cannot find last_sync_timestamp'
+        return false
+      end
+
+      query = { query: { last_sync_timestamp: last_sync,
+                         notifications_channel: channel },
+                headers: { 'accept':
+                            'application/vnd.system-notifications.1+json' } }
+
+      # rubocop:disable Lint/ShadowedException:
+      begin
+        @api_call = self.class.get('/api/system_notifications', query)
+
+        if @api_call.response.code.to_i != 200
+          @errors[:api_error] =
+            [@api_call.response.code.to_s, @api_call.response.message].join('-')
+        end
+      rescue SocketError, HTTParty::Error, StandardError => e
```

Comment:
@radiokills 
Until we handle all types of error in the same way, I would split this into many lines. What do you think?

Reply:
This makes sense :+1: